www/localPackage.ts

```diff
@@ -118,13 +117,18 @@ class LocalPackage extends Package implements ILocalPackage {
                     } else {
                         var invokeSuccessAndInstall = () => {
                             CodePushUtil.logMessage("Install succeeded.");
-                            installSuccess && installSuccess();
-                            /* no neeed for callbacks, the javascript context will reload */
-                            cordova.exec(() => { }, () => { }, "CodePush", "install", [deployDir.fullPath, installOptions.rollbackTimeout.toString(), installOptions.installMode.toString()]);
+                            if (installOptions.installMode === InstallMode.IMMEDIATE) {
+                                /* invoke success before navigating */
+                                installSuccess && installSuccess();
+                                /* no neeed for callbacks, the javascript context will reload */
+                                cordova.exec(() => { }, () => { }, "CodePush", "install", [deployDir.fullPath, installOptions.installMode.toString()]);
+                            } else {
+                                cordova.exec(() => { installSuccess && installSuccess(); }, () => { installError && installError(); }, "CodePush", "install", [deployDir.fullPath, installOptions.installMode.toString()]);
+                            }
                         };
 
                         var preInstallSuccess = () => {
-                            Sdk.reportStatus(AcquisitionStatus.DeploymentSucceeded);
+                            Sdk.reportStatus(AcquisitionStatus.DeploymentSucceeded, this);
```

Comment:
`reportStatus()` is now used to record that a user has either:
1. upgraded successfully to a new version or 
2. failed to upgrade to a new version (due to a crash triggering a rollback, or an install failure). 

Therefore, I think we should report DeploymentSucceeded during `notifyApplicationReady()`, if we see that the currentPackage has isFirstRun set to true, because that is the only time we can ascertain that an app has been installed successfully.

Also, we will need a new mechanism to find out that an app has been rolled back, and call reportStatus() there.